src/chainrulescore.jl

```diff
@@ -0,0 +1,129 @@
+
+@scalar_rule(elu(x, α), (delu(x, α), DoesNotExist()))
+function ChainRulesCore.rrule(
+    ::typeof(Base.Broadcast.broadcasted),
+    ::typeof(elu),
+    x::Numeric,
+    α::Numeric,
+)
+    broadcasted_elu_pullback(Δ) = (NO_FIELDS, NO_FIELDS, @thunk(delu.(x) .* Δ), DoesNotExist())
+    return elu.(x), broadcasted_elu_pullback
+end
+
+@scalar_rule(σ(x::Real), Ω * (1 - Ω))
+function ChainRulesCore.rrule(
+    ::typeof(Base.Broadcast.broadcasted),
+    ::typeof(σ),
+    x::Numeric,
+)
+    Ω = σ.(x)
+    broadcasted_σ_pullback(Δ) = (NO_FIELDS, NO_FIELDS, @thunk(Δ .* conj.(Ω .* (1 .- Ω))), DoesNotExist())
+    return Ω, broadcasted_σ_pullback
+end
+
+for softmax in [:softmax, :logsoftmax]
+    local ∇softmax = Symbol(:∇, softmax)
+    pullback = Symbol(softmax, :_pullback)
+
+    @eval function ChainRulesCore.rrule(::typeof($softmax), xs; dims=1)
+        $pullback(Δ) = (NO_FIELDS, @thunk($∇softmax(Δ, xs, dims=dims)))
+        return $softmax(xs; dims=dims), $pullback
+    end
+end
+
+for Dims in [:DenseConvDims, :DepthwiseConvDims, :PoolDims]
+    pullback = Symbol(Dims, :_pullback)
+    @eval function ChainRulesCore.rrule(::Type{$Dims}, args...; kwargs...)
+        $pullback(Δ) = (NO_FIELDS, ntuple(_ -> DoesNotExist(), length(args))...)
+        return $Dims(args...; kwargs...), $pullback
+    end
+end
+
+colmajor(x) = colmajor(MemoryLayout(typeof(x)), x)
+colmajor(_, x) = convert(Array, x)
+colmajor(::AbstractColumnMajor, x) = x
+
+for conv in [:conv, :depthwiseconv]
+    local ∇conv_data, ∇conv_filter = Symbol.(:∇, conv, [:_data, :_filter])
+    conv_pullback, ∇conv_data_pullback = Symbol.([conv, ∇conv_data], :_pullback)
+
+    @eval function ChainRulesCore.rrule(::typeof($conv), x, w, cdims; kw...)
+        function $conv_pullback(Δ)
+            Δ = colmajor(Δ)
+            return (
+                NO_FIELDS,
+                @thunk($∇conv_data(Δ, w, cdims, kw...)),
+                @thunk($∇conv_filter(x, Δ, cdims, kw...)),
+                DoesNotExist(),
+            )
+        end
+        return $conv(x, w, cdims; kw...), $conv_pullback
+    end
+
+    @eval function ChainRulesCore.rrule(::typeof($∇conv_data), x, w, cdims; kw...)
+        function $∇conv_data_pullback(Δ)
+            Δ = colmajor(Δ)
+            return (
+                NO_FIELDS,
+                @thunk($conv(Δ, w, cdims, kw...)),
+                @thunk($∇conv_filter(Δ, x, cdims, kw...)),
+                DoesNotExist(),
+            )
+        end
+        return $∇conv_data(x, w, cdims; kw...), $∇conv_data_pullback
+    end
+end
+
+for pool in [:maxpool, :meanpool]
+    ∇pool = Symbol(:∇, pool)
+    pullback = Symbol(pool, :_pullback)
+    @eval function ChainRulesCore.rrule(::typeof($pool), x, pdims::PoolDims; kw...)
+        Ω = maxpool(x, pdims; kw...)
+        $pullback(Δ) = (NO_FIELDS, @thunk($∇pool(Δ, Ω, x, pdims; kw...)), DoesNotExist())
+        return Ω, $pullback
+    end
+end
+
+function ChainRulesCore.rrule(::typeof(batched_mul), A, B)
```

Comment:
I think this should probably be restricted to `::AbstractArray{T,3}` for both. There are now also methods which accept `A::Matrix`, which is reshaped & sent to the 3-array method, and this rule applied directly to them won't work as `batched_adjoint` is then an error. 

Probably I should write some tests for their gradients since I'm to blame for the methods.

Reply:
Ok, I can add those type constraints